eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml

```diff
@@ -94,8 +94,10 @@
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientInstantiationCheck" files=".*[/\\]com[/\\]microsoft[/\\].*"/>
   <suppress checks="com.azure.tools.checkstyle.checks.ThrownClientLoggerCheck" files=".*[/\\]com[/\\]microsoft[/\\].*"/>
 
+  <!-- Custom checkstyle rules that don't apply to files under test or implementation package or samples -->
+  <suppress checks="com.azure.tools.checkstyle.checks.ExternalDependencyExposedCheck" files="(.*[/\\]src[/\\]test[/\\]java[/\\].*)|(.*[/\\]implementation[/\\].*)|(.*[/\\]samples[/\\].*)\.java"/>
```

Comment:
what do you say about having samples as another suppress configuration line.
Having it as an OR at the end and then a list of configs below make it easy to hide this suppression, IMO.  
Someone in the future would have it hard to find

Reply:
rearranged those suppressions for easier view and maintain.